norway_test.go

```diff
@@ -0,0 +1,130 @@
+package meteo_test
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/qba73/meteo"
+)
+
+func TestCreateNewMeteoClient(t *testing.T) {
+	t.Parallel()
+	var c *meteo.NorwayClient
+	resolver, err := meteo.NewWikipediaClient("User")
+	if err != nil {
+		t.Fatal(err)
+	}
+	c, err = meteo.NewNorwayClient(resolver)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_ = c
+}
+
+func TestCreateNewMeteoClientWithCustomUserAgent(t *testing.T) {
+	t.Parallel()
+	resolver, err := meteo.NewWikipediaClient("User")
+	if err != nil {
+		t.Fatal(err)
+	}
+	c, err := meteo.NewNorwayClient(
+		resolver,
+		meteo.WithUserAgent("CustomClient/1.0 https://customclient.com"),
+	)
+	if err != nil {
+		t.Fatalf("creating client with custom agent, %s\n", err)
+	}
+	want := "CustomClient/1.0 https://customclient.com"
+	got := c.UA
+	if want != got {
+		t.Errorf("want %q, got %q", want, got)
+	}
+}
+
+func TestCreateNewNorwayClientWithInvalidUserAgent(t *testing.T) {
+	t.Parallel()
+	resolver, err := meteo.NewWikipediaClient("User")
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = meteo.NewNorwayClient(resolver, meteo.WithUserAgent(""))
+	if err == nil {
+		t.Errorf("invalid user agent string should return error")
+	}
+}
+
+func TestGetForecast(t *testing.T) {
+	t.Parallel()
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/weatherapi/locationforecast/2.0/compact", func(rw http.ResponseWriter, r *http.Request) {
```

Comment:
Feels like this is testing a lot in one test!

A simpler test fixture might be easier to understand and there's less danger of bugs in the fixture.